language-support/ts/codegen/tests/build-and-lint.sh

```diff
@@ -2,7 +2,23 @@
 # Copyright (c) 2020 The DAML Authors. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-set -euo pipefail
+set -eo pipefail # Deliberately not set -u here.
+
+# It's sometimes useful to be able to run this test in a filesystem
+# that one can inspect.
+if [[ -z "$BUILD_AND_LINT_TMP_DIR" ]];
+then
+    TMP_DIR=$(mktemp -d)
+    cleanup() {
+        cd /
+        rm -rf $TMP_DIR
+    }
+    trap cleanup EXIT
+else
+  TMP_DIR="$BUILD_AND_LINT_TMP_DIR"
+  mkdir -p $TMP_DIR && rm -rf $TMP_DIR/*
```

Comment:
I think this would be easier to understand if you swap the order to `rm -rf $TMP_DIR && mkdir -p $TMP_DIR`.

Reply:
Ok!